xpra/server/auth/exec_auth.py

```diff
@@ -87,3 +96,54 @@ def command_timedout(self):
 
     def __repr__(self):
         return "exec"
+
+def add_cgi_headers(env : dict, request : WebSocketRequestHandler):
+    url = urlparse(request.path)
+    parts = url.path.rsplit('/', 1)
+    if len(parts)==1: # no '/', act as if there is a trailing '/'
+        parts = [parts, '']
+
+    # Not passing through any content
+    env['CONTENT_TYPE'] = ''
+    env['CONTENT_LENGTH'] = '0'
+
+    # see http.server.CGIHTTPRequestHandler.run_cgi()
+    # also RFC3875
+    env['SERVER_SOFTWARE'] = 'XPRA'
+    env['SERVER_PROTOCOL'] = request.protocol_version
+    env['GATEWAY_INTERFACE'] = 'CGI/1.1'
+    env['REQUEST_METHOD'] = request.command
+    env['PATH_INFO'] = parts[0]
+    env['PATH_TRANSLATED'] = parts[0] # TODO: window-ish path?
+    env['SCRIPT_NAME'] = parts[1]
+    if url.query:
+        env['QUERY_STRING'] = url.query
+    env['REMOTE_ADDR'] = request.client_address[0]
```

Comment:
Perhaps some of these values should be sanitized before being reflected back to the client.